atc/exec/get_step.go

```diff
@@ -168,68 +150,59 @@ func (step *GetStep) Run(ctx context.Context, state RunState) error {
 		return err
 	}
 
-	resourceInstance := resource.NewResourceInstance(
-		resource.ResourceType(step.plan.Type),
-		version,
+	processSpec := runtime.ProcessSpec{
+		Path:         "/opt/resource/in",
+		Args:         []string{resource.ResourcesDir("get")},
+		StdoutWriter: step.delegate.Stdout(),
+		StderrWriter: step.delegate.Stderr(),
+	}
+
+	resourceToGet := step.resourceFactory.NewResource(
 		source,
 		params,
-		resourceTypes,
-		resourceCache,
-		db.NewBuildStepContainerOwner(step.metadata.BuildID, step.planID, step.metadata.TeamID),
+		version,
 	)
 
-	chosenWorker, err := step.workerPool.FindOrChooseWorkerForContainer(
-		ctx,
-		logger,
-		resourceInstance.ContainerOwner(),
-		containerSpec,
-		workerSpec,
-		step.strategy,
-	)
-	if err != nil {
-		return err
-	}
+	containerOwner := db.NewBuildStepContainerOwner(step.metadata.BuildID, step.planID, step.metadata.TeamID)
 
+	// TODO: Starting should be emitted just before the get script is executed rather than at this point
 	step.delegate.Starting(logger)
 
-	versionedSource, err := step.resourceFetcher.Fetch(
+	getResult, err := step.workerClient.RunGetStep(
 		ctx,
```

Comment:
1. Is `RunXXXStep` a good naming?
2. Should we include those `RunXXXStep` in the interface `Client` or have a dedicated interface `ClientStepRunner` or something?